src/firefly/plans/regions_display.py

```diff
@@ -20,7 +20,89 @@ def setup_ui(self):
         raise NotImplementedError
 
 
-class RegionsDisplay(display.FireflyDisplay):
+class PlanDisplay(display.FireflyDisplay):
+    """Base class containing common functionality for basic plan window displays.
+    Should be subclassed to produce a usable display.
+    """
+
+    scan_time_changed = Signal(float)
+    total_time_changed = Signal(float)
+
+    def customize_ui(self):
+        self.ui.run_button.clicked.connect(self.queue_plan)
+        if hasattr(self.ui, "spinBox_repeat_scan_num"):
```

Comment:
I think this belongs in the individual child classes instead of the parent classes. Having a parent class make decisions based on things that *might* be done by the child classes can cause problems.

For example, what if the child class wants to use this spin box differently, or call it something else, or it's not actually a Qt Signal? The child class has no way to do this without breaking the inheritance chain for this method.